src/formats/spec.ts

```diff
@@ -0,0 +1,119 @@
+import type { Result as AxeResult, NodeResult } from 'axe-core';
+import chalk from 'chalk';
+import indent from 'indent-string';
+import dedent from 'ts-dedent';
+import { Result } from '../Result';
+import { SuiteEmitter } from '../Suite';
+
+/**
+ * Output test suite information similarly to Mocha's "spec" formatter.
+ */
+export function format(emitter: SuiteEmitter, print = console.log, colors = new chalk.Instance()): void {
```

Comment:
Note the arguments for `print` and `colors`, instead of directly using `console.log` and `chalk`. That makes testing much easier.